joint_trajectory_controller/src/trajectory.cpp

```diff
@@ -64,6 +90,7 @@ bool Trajectory::sample(
   TrajectoryPointConstIter & start_segment_itr, TrajectoryPointConstIter & end_segment_itr)
 {
   THROW_ON_NULLPTR(trajectory_msg_)
+  // TODO(anyone): this shouldn't be initialized at runtime
   output_state = trajectory_msgs::msg::JointTrajectoryPoint();
```

Comment:
I think this line is OK, actually, since the size of  `trajectory_msgs::msg::JointTrajectoryPoint();` is known. This line (probably) won't add any overhead. See https://stackoverflow.com/a/3835902

What will hurt runtime is a few lines down, when `output_state` is resized.

`output_state = state_before_traj_msg_;`

I'm glad you're thinking about this though!

The best improvement I can think of is:

```suggestion
  `trajectory_msgs::msg::JointTrajectoryPoint& output_state;`
```